app/Main.hs

```diff
@@ -52,6 +53,12 @@ contracts = ContractRegistry $ Map.fromList
     , ciStorageParser = Just baseDaoStorageParser
     , ciStorageNotes = Nothing
     }
+  , "GameDAO" ?:: ContractInfo
+    { ciContract = GameDAO.gameDaoContract
+    , ciIsDocumented = True
+    , ciStorageParser = Nothing
```

Comment:
I think it's better to provide a parser here, it should be similar to `baseDaoStorageParser` AFAIU.

Reply:
I have updated this, as well as adding tests for GameDAO. I'll wait to see if it passes the network test.

Reply:
Unfortunately, the origination failed due to the large operation size. We probably have to wait for #40 to be merged.

Reply:
I've rebased this MR with the latest master. The tests are still failing due to this error:

```
tezos-client unexpectedly failed with error code 1. Stdout:

          Node is bootstrapped, ready for injecting operations.

          

          Stderr:

          Fatal error:

            The operation will burn ꜩ13.403 which is higher than the configured burn cap (ꜩ13.389).

             Use `--burn-cap 13.403` to emit this operation.
```

This is a new error, I'm not sure how to fix this.

Reply:
:fffuuu:

Most likely it's a consequence of https://gitlab.com/morley-framework/morley/-/issues/288. It happens when you have an address constant in contract code. And apparently that's what we are doing in our decision lambda.

It seems quite hard to fix properly in `morley`, to me it looks more like a bug in Tezos. Anyway, there are two options:
1. Fix/workaround this problem in `morley`. I'll think about it tomorrow, probably there is an easy workaround.
2. Fetch `consumerAddr` from the storage. Currently we can't extend the storage with arbitrary data, but it was suggested by @Martoon-00 and probably we'll implement it soon. Then it will be possible to have `consumerAddr` in storage instead of contract code. Another option that is already possible is to store `consumerAddr` inside each proposal, which might be weird, but probably doesn't matter for illustration purposes.

Reply:
I store `consumerAddr` in the proposal itself, and now the tests are passed now. I've also made other changes according to your review, you can check this.

Reply:
Oh, cool, then I think we can merge it as is now. Maybe later we'll move consumer address somewhere else.